types/node/index.d.ts

```diff
@@ -2354,7 +2354,7 @@ declare module "net" {
         unref(): void;
         ref(): void;
 
-        remoteAddress: string;
```

Comment:
@amcasey I think you're right. it seems that remoteFamily and remotePort should be returned with undefined if remoteAddress is undefined.

https://github.com/nodejs/node/blob/master/lib/net.js#L659-L669
we could see from lines here, they all return by `_getpeername`. If there's no `handle` in socket,` _getpeername` will return empty object, hence return undefined in remoteAddress, remoteFamily, remotePort

@Flarna  @kazuyamamoto what do you think?